integration-tests/test_client.py

```diff
@@ -0,0 +1,99 @@
+import glob
+import logging
+import os
+import subprocess
+from os.path import exists
+
+import pytest
+
+logger = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_client_files_permission(insights_client):
+    """Verify that permission for following files is 0644:
+    - /etc/insights-client/.lastupload
+    - /etc/insights-client/.last-upload.results only created in legacy mode which
+    will be deprecated soon, hence skipping
+
+    Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1924990
+    """
+    file_ = "/etc/insights-client/.lastupload"
+    if exists(file_):
+        os.remove(file_)  # performing a cleanup before test
+
+    insights_client.register()
+    assert insights_client.is_registered
+    assert oct(os.stat(file_).st_mode)[-3:] == "644"
+
+
+@pytest.mark.parametrize(
+    "file",
+    [
+        "/etc/insights-client/insights-client.motd",
+        "/etc/insights-client/insights-client.conf",
+        "/usr/lib/systemd/system/insights-client-boot.service",
+        "/usr/lib/systemd/system/insights-client-results.path",
+        "/usr/lib/systemd/system/insights-client-results.service",
+        "/usr/lib/systemd/system/insights-client.service",
+        "/usr/lib/systemd/system/insights-client.timer",
+        "/usr/share/doc/insights-client",
+        "/usr/share/doc/insights-client/file-content-redaction.yaml.example",
+        "/usr/share/doc/insights-client/file-redaction.yaml.example",
+        "/usr/share/man/man5/insights-client.conf.5.gz",
+        "/usr/share/man/man8/insights-client.8.gz",
+    ],
+)
+def test_client_rpm_mandatory_files(file):
+    """
+    Verify the existence of mandatory files for insights-client rpm
+    """
+    if (
+        pytest.rhel_major_version in ("6", "7") or pytest.centos_major_version == "7"
+    ) and "motd" in file:
+        pytest.skip("motd feature not enabled in RHEL6/7 and centos7 RPMs")
+
+    elif pytest.rhel_major_version in ("6") and "systemd" in file:
+        pytest.skip("No systemd files in RHEL6 RPMs")
+
+    else:
+        cmd = ["rpm", "-ql", "insights-client"]
+        rpm_ql_output = subprocess.run(cmd, capture_output=True, text=True)
+        assert file in rpm_ql_output.stdout
```

Comment:
while this works, it is a bit inefficient, as it will run `rpm -ql` for every combination of this parametrized test case

what I'd do instead is to save the split lines of `rpm -ql` as fixture for the test, so it is run only once

Reply:
ok.